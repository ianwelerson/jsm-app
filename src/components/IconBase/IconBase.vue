<script setup lang="ts">
import { computed } from 'vue'
import { defineAsyncComponent } from 'vue'

export interface IconBaseProps {
  color?: string
  height: number
  name: string
  width?: number
}

const props = withDefaults(defineProps<IconBaseProps>(), {
  color: '#ffffff',
})

const iconHeight = computed(() => {
  return numberToSize(props.height)
})

const iconWidth = computed(() => {
  return numberToSize(props.width ?? props.height)
})

const IconComponent = defineAsyncComponent(
  () => import(`./IconOptions/${props.name}.vue`)
)

const numberToSize = (number: number) => {
  return `${number}px`
}
</script>

<template>
  <div data-testid="icon-base" class="icon-base">
    <IconComponent />
  </div>
</template>

<style lang="scss" scoped>
.icon-base {
  color: v-bind('props.color');
  height: v-bind('iconHeight');
  width: v-bind('iconWidth');
}
</style>
