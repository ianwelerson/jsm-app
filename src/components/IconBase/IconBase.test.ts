import { render } from '@testing-library/vue'
import IconBase from '@/components/IconBase/IconBase.vue'

const propsExample = {
  color: '#000000',
  height: 14,
  name: 'IconMenu',
}

describe('IconBase', () => {
  it('should render icon with all props', async () => {
    const { getByTestId } = render(IconBase, {
      props: {
        ...propsExample,
        width: 14,
      },
    })

    await vi.dynamicImportSettled()

    getByTestId('icon-base')
  })

  it('should match the snapshot', async () => {
    const { html } = render(IconBase, {
      props: {
        ...propsExample,
      },
    })

    await vi.dynamicImportSettled()

    expect(html()).toMatchSnapshot()
  })
})
