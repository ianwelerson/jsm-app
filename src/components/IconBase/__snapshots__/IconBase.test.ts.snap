// Vitest Snapshot v1

exports[`IconBase > should match the snapshot 1`] = `
"<div data-testid=\\"icon-base\\" class=\\"icon-base\\" data-v-2336b232=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"100%\\" width=\\"100%\\" viewBox=\\"0 0 48 48\\" data-v-2336b232=\\"\\">
    <path d=\\"M6 36v-3h36v3Zm0-10.5v-3h36v3ZM6 15v-3h36v3Z\\" fill=\\"currentColor\\"></path>
  </svg></div>"
`;
